packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -558,12 +558,8 @@ add (const Scalar& alpha,
   {
     if(params->isParameter("Call fillComplete") && !params->get<bool>("Call fillComplete"))
     {
-      if(A.getRowMap()->getComm()->getRank() == 0)
-      {
-        std::cout << "WARNING: Tpetra::MatrixMatrix::add(): 'Call fillComplete' was explicitly set to false,\n";
-        std::cout << "but this version of add() always calls fillComplete() on the sum (C). Use the other version\n";
-        std::cout << "of add() (that works on a newly constructed C) to get a non-fillComplete matrix.\n";
-      }
+      throw std::invalid_argument("Tpetra::MatrixMatrix::add(): this version of add() always calls fillComplete\n"
```

Comment:
"parameter list" (spelling).  Also, prefer the `TEUCHOS_TEST_FOR_EXCEPTION` macro as a point at which we can inject a breakpoint in case this exception should come up.